src/data/equiv/basic.lean

```diff
@@ -1723,10 +1712,25 @@ protected def powerset {α} (S : set α) : 𝒫 S ≃ set S :=
 
 end set
 
+/-- If `f : α → β` is an injective function, then domain `α` is equivalent to the range of `f`. -/
+@[simps apply]
+noncomputable def of_injective {α β} (f : α → β) (hf : injective f) : α ≃ _root_.set.range f :=
+equiv.set.range_of_left_inverse f
```

Comment:
This name would be better without the `set.range_`, which was there only to match the definition you've now removed